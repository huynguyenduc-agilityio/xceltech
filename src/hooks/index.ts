export * from './useToast';
export * from './useConfirm';
export * from './useAuth';
export * from './useUser';
export * from './useLeave';
export * from './useAccount';
export * from './usePagination';
export * from './useNotification';
export * from './useJob';
export * from './useDocument';
export * from './useEducation';
export * from './useGuarantor';
export * from './useFamily';
export * from './useFinancial';
