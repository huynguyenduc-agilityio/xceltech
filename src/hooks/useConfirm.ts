import { useConfirmDialogStore } from '@/stores';

export const useConfirm = () => {
  return useConfirmDialogStore((state) => state.confirm);
};
