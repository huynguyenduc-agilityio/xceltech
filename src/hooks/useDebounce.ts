import { useCallback, useRef } from 'react';

export const useDebounce = <TParam = unknown>(
  callback: (...args: TParam[]) => void,
  delay = 600,
) => {
  const refTime = useRef<NodeJS.Timeout | null>(null);

  const debouncedFn = useCallback(
    (...args: TParam[]) => {
      if (refTime.current) clearTimeout(refTime.current);
      refTime.current = setTimeout(() => {
        callback(...args);
      }, delay);
    },
    [callback, delay],
  );

  return debouncedFn;
};
