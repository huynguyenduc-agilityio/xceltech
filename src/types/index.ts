export * from './status';
export * from './employee';
export * from './table';
export * from './auth';
export * from './form';
export * from './tab';
export * from './user';
export * from './queryKey';
export * from './leave';
export * from './filter';
export * from './error';
export * from './notification';
export * from './profile';
