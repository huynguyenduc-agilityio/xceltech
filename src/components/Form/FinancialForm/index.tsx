import { useForm } from 'react-hook-form';
import { zodResolver } from '@hookform/resolvers/zod';

// Constants
import { MESSAGES } from '@/constants';

// Types
import {
  FinancialFormValues,
  IEmployeeFinancialInfo,
  MutationType,
  ToastStatus,
} from '@/types';

//Hooks
import { useFinancialMutation, useToast } from '@/hooks';

// Components
import { Button, Form, FormField, TextField } from '@/components';

// Utils
import { financialSchema } from '@/utils';

export interface IFinancialForm {
  mode: MutationType;
  initialValues?: Partial<IEmployeeFinancialInfo>;
  onBack?: () => void;
}

const FinancialForm = ({
  initialValues = {},
  mode,
  onBack,
}: IFinancialForm) => {
  const { toast } = useToast();
  const { handleFinancialMutation, isFinancialMutationLoading } =
    useFinancialMutation({
      type: mode,
    });

  const { accountName = '', bankName = '', accountNo = 0 } = initialValues;
  const defaultValues: IEmployeeFinancialInfo = {
    bankName,
    accountNo,
    accountName,
  };

  const form = useForm<FinancialFormValues>({
    mode: 'onBlur',
    reValidateMode: 'onChange',
    resolver: zodResolver(financialSchema),
    defaultValues,
  });

  const {
    control,
    handleSubmit,
    reset,
    getValues,
    formState: { isValid, isDirty, isSubmitting },
  } = form;

  const isCreate = mode === MutationType.Create;

  const onSubmit = async (data: IEmployeeFinancialInfo) => {
    const payload = {
      ...data,
      ...(!isCreate && initialValues?.id ? { id: initialValues.id } : {}),
    };

    try {
      await handleFinancialMutation(
        payload as unknown as IEmployeeFinancialInfo,
      );

      toast({
        status: ToastStatus.Success,
        title: isCreate
          ? MESSAGES.COMMON.ADD_SUCCESS('Financial')
          : MESSAGES.COMMON.UPDATE_SUCCESS('Financial'),
      });
      onBack?.();

      reset(initialValues);
    } catch {
      toast({
        status: ToastStatus.Error,
        title: isCreate
          ? MESSAGES.COMMON.ADD_FAILED('Financial')
          : MESSAGES.COMMON.UPDATE_FAILED('Financial'),
      });

      reset(getValues());
    }
  };

  const disableSubmit = !isDirty || !isValid || isSubmitting;

  return (
    <Form {...form}>
      <form onSubmit={handleSubmit(onSubmit)}>
        <div className="h-[741px] flex flex-col justify-between">
          <div className="w-full  text-black-default space-y-10">
            <FormField
              control={control}
              name="bankName"
              render={({ field, fieldState: { error } }) => (
                <TextField
                  label="Bank Name"
                  variant="primary"
                  className="space-y-5"
                  labelClassName="text-md"
                  inputClassName="h-[68px] px-12 rounded-[15px]"
                  placeholder="Enter your bank name"
                  {...field}
                  errorMessage={error?.message}
                />
              )}
            />
            <div className="flex items-start gap-16">
              <FormField
                control={control}
                name="accountNo"
                render={({ field, fieldState: { error } }) => (
                  <TextField
                    type="number"
                    label="Account No"
                    variant="primary"
                    className="space-y-5"
                    labelClassName="text-md"
                    inputClassName="h-[68px] px-12 rounded-[15px]"
                    placeholder="Enter your account number"
                    {...field}
                    errorMessage={error?.message}
                  />
                )}
              />
              <FormField
                control={control}
                name="accountName"
                render={({ field, fieldState: { error } }) => (
                  <TextField
                    label="Account Name"
                    variant="primary"
                    className="space-y-5"
                    labelClassName="text-md"
                    inputClassName="h-[68px] px-12 rounded-[15px]"
                    placeholder="Enter your account name"
                    {...field}
                    errorMessage={error?.message}
                  />
                )}
              />
            </div>
          </div>

          <div className="w-full h-[70px] flex items-center justify-center">
            <Button
              type="submit"
              className="w-[587px] bg-green-primary rounded-lg"
              isLoading={isFinancialMutationLoading}
              disabled={disableSubmit}
            >
              {isCreate ? 'Add Account Details' : 'Update Account Details'}
            </Button>
          </div>
        </div>
      </form>
    </Form>
  );
};

export default FinancialForm;
