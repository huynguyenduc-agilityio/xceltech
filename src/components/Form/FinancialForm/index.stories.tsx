import type { Meta, StoryObj } from '@storybook/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

import { MutationType } from '@/types';

// Components
import FinancialForm from '.';

const queryClient = new QueryClient();

const meta = {
  title: 'Components/FinancialForm',
  component: FinancialForm,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  render: (args) => (
    <QueryClientProvider client={queryClient}>
      <div className="flex items-center justify-center w-screen py-20 bg-blue-light">
        <div className="bg-white w-[988px] rounded-[15px]">
          <div className="p-10">
            <FinancialForm {...args} />
          </div>
        </div>
      </div>
    </QueryClientProvider>
  ),
} satisfies Meta<typeof FinancialForm>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    mode: MutationType.Create,
  },
};
