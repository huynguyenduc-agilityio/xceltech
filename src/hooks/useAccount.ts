import { useQuery } from '@tanstack/react-query';

// Constants
import { accountsQueryKeys } from '@/constants';

// Types
import { IInfoUser } from '@/types';

// Services
import { getAccounts } from '@/services';

export const useGetLeaveAccounts = (userId: string) => {
  const { data: leaveAccounts, isFetching: isLeaveAccountsLoading } = useQuery({
    queryKey: [...accountsQueryKeys.list()] as const,
    queryFn: getAccounts,
    select: (data) =>
      data
        ?.filter(({ id }: { id: string }) => id !== userId)
        .map(({ id, firstName, lastName }: IInfoUser) => ({
          value: id,
          label: `${firstName} ${lastName}`,
        })),
  });

  return { leaveAccounts, isLeaveAccountsLoading };
};
