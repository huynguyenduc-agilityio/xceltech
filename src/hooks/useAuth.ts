import { useNavigate, useLocation } from 'react-router-dom';
import { useMutation } from '@tanstack/react-query';

//Types
import { IActiveUser } from '@/types';

// Constants
import {
  ADMIN_PAGE,
  AUTHENTICATION_PAGE,
  RoleAuthentication,
  USER_PAGE,
} from '@/constants';

// Services
import { activateAccount, loginUser, registerUser } from '@/services';

// Stores
import { useUserActions } from '@/stores';

export const useLogin = () => {
  const navigate = useNavigate();
  const location = useLocation();
  const { setUser } = useUserActions();

  const {
    mutateAsync: handleLoginUser,
    isPending: isLoginLoading,
    isError: isLoginError,
  } = useMutation({
    mutationFn: loginUser,
    onSuccess: (data) => {
      // Save data in local storage
      if (data) {
        const path = location.pathname;
        const role = data.user.role;
        const isAdmin =
          role === RoleAuthentication.Admin &&
          path === AUTHENTICATION_PAGE.ADMIN_SIGN_IN;

        const isUser =
          [RoleAuthentication.Employee, RoleAuthentication.Candidate].includes(
            role as RoleAuthentication,
          ) && path === AUTHENTICATION_PAGE.USER_SIGN_IN;

        if (isAdmin) {
          setUser(data);
          navigate(ADMIN_PAGE.DASHBOARD);
        }

        if (isUser) {
          setUser(data);
          navigate(USER_PAGE.DASHBOARD);
        }
      }
    },
  });

  return { isLoginLoading, isLoginError, handleLoginUser };
};

export const useRegister = () => {
  const { mutateAsync: handleRegisterUser, isPending: isRegisterLoading } =
    useMutation({
      mutationFn: registerUser,
    });

  return {
    handleRegisterUser,
    isRegisterLoading,
  };
};

export const useActivateAccount = ({ uidb64, token }: IActiveUser) => {
  const { mutateAsync: handleActivateUser, isPending: isActivateLoading } =
    useMutation({
      mutationFn: () => activateAccount({ uidb64, token }),
    });

  return {
    handleActivateUser,
    isActivateLoading,
  };
};
