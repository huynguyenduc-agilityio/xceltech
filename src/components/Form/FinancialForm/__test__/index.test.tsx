import { fireEvent, render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

import { MutationType } from '@/types';

// Components
import FinancialForm from '..';

// Hooks
import { useFinancialMutation, useToast } from '@/hooks';

const queryClient = new QueryClient();
const mockHandleFinancialMutation = jest.fn();
const mockHandleFinancialMutationError = jest
  .fn()
  .mockRejectedValue(new Error('Network error'));
const mockToast = jest.fn();
const mockDataEdit = {
  id: '1',
  bankName: 'Test Bank',
  accountNo: 1234567890,
  accountName: 'John Doe',
};

jest.mock('@/hooks', () => ({
  useToast: jest.fn(),
  useFinancialMutation: jest.fn(),
}));

const renderComponent = () => {
  return render(
    <QueryClientProvider client={queryClient}>
      <FinancialForm mode={MutationType.Edit} />
    </QueryClientProvider>,
  );
};

describe('FinancialForm Component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    (useToast as jest.Mock).mockReturnValue({ toast: mockToast });
    (useFinancialMutation as jest.Mock).mockReturnValue({
      isFamilyMutationLoading: false,
      handleFinancialMutation: mockHandleFinancialMutation,
    });
  });

  it('should match snapshot', () => {
    const container = renderComponent();

    expect(container).toMatchSnapshot();
  });

  it('handle input value change and form add submit', async () => {
    const { getByPlaceholderText, getByRole } = render(
      <QueryClientProvider client={queryClient}>
        <FinancialForm mode={MutationType.Create} />
      </QueryClientProvider>,
    );
    const bankName = getByPlaceholderText('Enter your bank name');
    const accountNo = getByPlaceholderText('Enter your account number');
    const accountName = getByPlaceholderText('Enter your account name');
    const submitButton = getByRole('button', {
      name: 'Add Account Details',
    });

    fireEvent.change(bankName, { target: { value: 'Test Bank' } });
    fireEvent.change(accountNo, { target: { value: 1234567890 } });
    fireEvent.change(accountName, { target: { value: 'John Doe' } });

    await userEvent.click(submitButton);

    await waitFor(() => {
      expect(mockHandleFinancialMutation).toHaveBeenCalled();
    });
  });

  it('handle input value change and form edit submit', async () => {
    const { getByPlaceholderText, getByRole } = render(
      <QueryClientProvider client={queryClient}>
        <FinancialForm mode={MutationType.Edit} initialValues={mockDataEdit} />
      </QueryClientProvider>,
    );
    const bankName = getByPlaceholderText('Enter your bank name');
    const accountNo = getByPlaceholderText('Enter your account number');
    const accountName = getByPlaceholderText('Enter your account name');
    const submitButton = getByRole('button', {
      name: 'Update Account Details',
    });

    fireEvent.change(bankName, { target: { value: 'Test Bank' } });
    fireEvent.change(accountNo, { target: { value: 1234567891 } });
    fireEvent.change(accountName, { target: { value: 'John Doe' } });

    await userEvent.click(submitButton);

    await waitFor(() => {
      expect(mockHandleFinancialMutation).toHaveBeenCalled();
    });
  });

  it('handles form add submit error gracefully', async () => {
    (useFinancialMutation as jest.Mock).mockReturnValue({
      isLoading: false,
      handleFinancialMutation: mockHandleFinancialMutationError,
    });

    const { getByPlaceholderText, getByRole } = render(
      <QueryClientProvider client={queryClient}>
        <FinancialForm mode={MutationType.Create} />
      </QueryClientProvider>,
    );
    const bankName = getByPlaceholderText('Enter your bank name');
    const accountNo = getByPlaceholderText('Enter your account number');
    const accountName = getByPlaceholderText('Enter your account name');
    const submitButton = getByRole('button', {
      name: 'Add Account Details',
    });

    fireEvent.change(bankName, { target: { value: 'Test Bank' } });
    fireEvent.change(accountNo, { target: { value: 1234567890 } });
    fireEvent.change(accountName, { target: { value: 'John Doe' } });

    await userEvent.click(submitButton);

    await waitFor(() => {
      expect(mockToast).toHaveBeenCalled();
    });
  });

  it('handles form edit submit error gracefully', async () => {
    (useFinancialMutation as jest.Mock).mockReturnValue({
      isLoading: false,
      handleFinancialMutation: mockHandleFinancialMutationError,
    });

    const { getByPlaceholderText, getByRole } = render(
      <QueryClientProvider client={queryClient}>
        <FinancialForm mode={MutationType.Edit} initialValues={mockDataEdit} />
      </QueryClientProvider>,
    );
    const bankName = getByPlaceholderText('Enter your bank name');
    const accountNo = getByPlaceholderText('Enter your account number');
    const accountName = getByPlaceholderText('Enter your account name');
    const submitButton = getByRole('button', {
      name: 'Update Account Details',
    });

    fireEvent.change(bankName, { target: { value: 'Test Bank' } });
    fireEvent.change(accountNo, { target: { value: 1234567891 } });
    fireEvent.change(accountName, { target: { value: 'John Doe' } });

    await userEvent.click(submitButton);

    await waitFor(() => {
      expect(mockToast).toHaveBeenCalled();
    });
  });

  it('should show error message fields', async () => {
    const { getByPlaceholderText, getByText } = renderComponent();
    const bankName = getByPlaceholderText('Enter your bank name');
    const accountNo = getByPlaceholderText('Enter your account number');
    const accountName = getByPlaceholderText('Enter your account name');

    fireEvent.change(bankName, { target: { value: '' } });
    fireEvent.change(accountNo, { target: { value: '' } });
    fireEvent.change(accountName, { target: { value: '' } });

    fireEvent.blur(bankName);
    fireEvent.blur(accountNo);
    fireEvent.blur(accountName);

    await waitFor(() => {
      expect(getByText('Bank Name is required!')).toBeInTheDocument();
      expect(getByText('Account No is required!')).toBeInTheDocument();
      expect(getByText('Account Name is required!')).toBeInTheDocument();
    });
  });
});
