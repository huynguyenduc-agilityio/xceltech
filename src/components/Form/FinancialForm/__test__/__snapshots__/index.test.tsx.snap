// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinancialForm Component should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="h-[741px] flex flex-col justify-between"
        >
          <div
            class="w-full  text-black-default space-y-10"
          >
            <div
              class="grid w-full gap-2"
            >
              <div
                class="w-full space-y-5"
              >
                <label
                  class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                  id="Bank Name-label"
                >
                  Bank Name
                </label>
                <input
                  aria-describedby=":r0:-form-item-description"
                  aria-invalid="false"
                  aria-labelledby="Bank Name-label"
                  autocomplete="off"
                  class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                  data-testid="input-element"
                  id=":r0:-form-item"
                  name="bankName"
                  placeholder="Enter your bank name"
                  value=""
                />
              </div>
            </div>
            <div
              class="flex items-start gap-16"
            >
              <div
                class="grid w-full gap-2"
              >
                <div
                  class="w-full space-y-5"
                >
                  <label
                    class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                    id="Account No-label"
                  >
                    Account No
                  </label>
                  <input
                    aria-describedby=":r1:-form-item-description"
                    aria-invalid="false"
                    aria-labelledby="Account No-label"
                    autocomplete="off"
                    class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                    data-testid="input-element"
                    id=":r1:-form-item"
                    name="accountNo"
                    placeholder="Enter your account number"
                    type="number"
                    value="0"
                  />
                </div>
              </div>
              <div
                class="grid w-full gap-2"
              >
                <div
                  class="w-full space-y-5"
                >
                  <label
                    class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                    id="Account Name-label"
                  >
                    Account Name
                  </label>
                  <input
                    aria-describedby=":r2:-form-item-description"
                    aria-invalid="false"
                    aria-labelledby="Account Name-label"
                    autocomplete="off"
                    class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                    data-testid="input-element"
                    id=":r2:-form-item"
                    name="accountName"
                    placeholder="Enter your account name"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="w-full h-[70px] flex items-center justify-center"
          >
            <button
              class="font-productsans inline-flex items-center justify-center whitespace-nowrap text-lg font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed text-white h-[70px] py-6 w-[587px] bg-green-primary rounded-lg"
              disabled=""
              type="submit"
            >
              Update Account Details
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="h-[741px] flex flex-col justify-between"
      >
        <div
          class="w-full  text-black-default space-y-10"
        >
          <div
            class="grid w-full gap-2"
          >
            <div
              class="w-full space-y-5"
            >
              <label
                class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                id="Bank Name-label"
              >
                Bank Name
              </label>
              <input
                aria-describedby=":r0:-form-item-description"
                aria-invalid="false"
                aria-labelledby="Bank Name-label"
                autocomplete="off"
                class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                data-testid="input-element"
                id=":r0:-form-item"
                name="bankName"
                placeholder="Enter your bank name"
                value=""
              />
            </div>
          </div>
          <div
            class="flex items-start gap-16"
          >
            <div
              class="grid w-full gap-2"
            >
              <div
                class="w-full space-y-5"
              >
                <label
                  class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                  id="Account No-label"
                >
                  Account No
                </label>
                <input
                  aria-describedby=":r1:-form-item-description"
                  aria-invalid="false"
                  aria-labelledby="Account No-label"
                  autocomplete="off"
                  class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                  data-testid="input-element"
                  id=":r1:-form-item"
                  name="accountNo"
                  placeholder="Enter your account number"
                  type="number"
                  value="0"
                />
              </div>
            </div>
            <div
              class="grid w-full gap-2"
            >
              <div
                class="w-full space-y-5"
              >
                <label
                  class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                  id="Account Name-label"
                >
                  Account Name
                </label>
                <input
                  aria-describedby=":r2:-form-item-description"
                  aria-invalid="false"
                  aria-labelledby="Account Name-label"
                  autocomplete="off"
                  class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                  data-testid="input-element"
                  id=":r2:-form-item"
                  name="accountName"
                  placeholder="Enter your account name"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="w-full h-[70px] flex items-center justify-center"
        >
          <button
            class="font-productsans inline-flex items-center justify-center whitespace-nowrap text-lg font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed text-white h-[70px] py-6 w-[587px] bg-green-primary rounded-lg"
            disabled=""
            type="submit"
          >
            Update Account Details
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
